### DISCUSSION TITLE
Add combined mode for chat

### DISCUSSION THREAD

**Discussion by DraconicCode** [Chatterino/chatterino2#4056] [Ideas]
New idea: allow multiple chats to be combined into one where messages you send can either go to both or one of your choice, and messages received will appear in the same space for both chats

**Comment by pajlada** [comment]
Could you describe a bit more in detail how you envision this functionality looking or behaving to give folks a base idea?

**Reply by DraconicCode** [reply]
A quick mockup I made:
![drawing-1](https://user-images.githubusercontent.com/108132722/195900930-d59b61b0-2b24-4f3d-849a-6a326796c932.png)

**Reply by ref4time** [reply]
@dieserRobin, I think this is something for you.

**Comment by JCRouzer29** [comment]
Reviving this feature request.

Hoping one is able to make use of the way shared chat on Twitch functions to add multiple channels' chats into a single Chatterino channel. Using the profile picture of the source channel to distinguish their origin.

For an initial implementation, comma separating the channels in the channel dialog box. Also I can imagine having a requirement of the chats being opened separately akin to loading the 7tv/bttv/ffz emotes to display in shared chat. 

<img width="474" height="472" alt="Dialog" src="https://github.com/user-attachments/assets/8d7a5f56-3770-4621-b4c1-4f56fad802ce" />
<img width="792" height="905" alt="Visual Example" src="https://github.com/user-attachments/assets/7c230dc8-4c90-4ca4-b1d9-eae20a5f453f" />

**Reply by Nerixyz** [reply]
I implemented something like this recently. The main issue was getting the badges in the right position. One constraint I set for myself was that the channel would not need to clone messages.

One idea how this could be resolved: Put a reference to the channel itself in the messages and add a "channel indicator" element to messages. That could show both the channel name and a badge that it gets from the channel.

**Reply by Yusuf-joe** [reply]
Can i suggest another related feature? 

The ability to split combined shared chats into 2 separate splits and/or show only one of them.

Maybe through filters?

That way streamers and viewers that want to read only one of the chats combined are able to do so, and those who want to combine 2 chats that arent combined by twitch are able to do so.

But idk how feasible that is.

Good luck.
